racket/collects/racket/promise.rkt

```diff
@@ -223,3 +225,31 @@
                               (cons '#:work-while #'(system-idle-evt))
                               (cons '#:tick       #'0.2)
                               (cons '#:use        #'0.12))))
+
+(define-syntaxes (for/concurrent for*/concurrent)
+  (let ()
+    (define ((make-transformer for/fold/derived-stx [orig-stx #f]) stx)
+      (syntax-case stx ()
+        [(_ #:group group-expr clauses body ...)
+         (let ([stx (or orig-stx stx)])
+           (with-syntax ([this-syntax stx]
+                         [for/fold/derived for/fold/derived-stx]
+                         [((pre-body ...)
+                           (post-body ...))
+                          (split-for-body stx #'(body ...))])
+             (syntax/loc stx
+               (let ([group group-expr])
+                 (for/fold/derived this-syntax
+                   ([promises null] #:result (map force (reverse promises)))
+                   clauses
+                   pre-body ...
+                   (cons (delay/thread*
+                          #:group group
+                          (let () post-body ...))
+                         promises))))))]
+        [(_ clauses body ...)
+         ((make-transformer for/fold/derived-stx stx)
+          #'(fc #:group (make-thread-group) clauses body ...))]))
```

Comment:
Where is `fc` bound?

Reply:
It's not bound anywhere because the macro ignores it. It just needs to be there to match the pattern (and this is how a few of the other `for` macros (like `for/vector`) do recursion).